### PR TITLE
Proposal to circumvent 50 tag ASG limit for EKS ManagedNodegroups

### PR DESCRIPTION
**Pull request by MyannaHarris**
This proposal is related to this issue that AWS EKS has open: https://github.com/aws/containers-roadmap/issues/724

The proposal was originally in kubernetes/enhancements. See the previous feedback cycle here: https://github.com/kubernetes/enhancements/pull/2561

@gjtempleton 
@jaypipes

### CODE REVIEW

**Comment by gjtempleton** [comment]
Nice work! All of my comments from the previous discussion in kubernetes/enhancements#2561 have been answered/addressed, so:
/lgtm

File: cluster-autoscaler/proposals/circumvent-tag-limit-aws.md [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+- [TC3] Test pulling values from cache and calling DescribeNodegroup API if no data exists
+- [TC4] Test pulling values from cache and calling DescribeNodegroup API if TTL has expired
+- [TC5] Test missing permissions and falling back to just ASG tags
+- [TC6] Test duplicate label keys - selecting the value from ASG tags response
+
+## Risks
+
+Latency and throttling:
+
+By default, Cluster Autoscaler runs every 10 seconds. Our best practices documentation notes that this short interval can cause throttling because Cluster Autoscaler already makes AWS API calls during each loop. Our documentation recommends that customers increase the interval, so adding this API shouldn’t cause latency problems for customers. ([EKS Best Practices][]) Also, the API call will only happen for EKS ManagedNodegroups. If this increase in latency is too much for even one run of the loop, we will look into moving the API calls into parallel goroutines.
```
Comment (elmiko) [inline cluster-autoscaler/proposals/circumvent-tag-limit-aws.md:130]: i'm just curious if there is precedent for using parallel goroutines in cluster-autoscaler to do similar types of API calls?

**Review by elmiko — COMMENTED** [review]
i am not intimately familiar with the AWS provider code, but this proposal seems reasonable to me. i have one minor question, but it's more to help my understanding.

**Comment by MaciekPytel** [comment]
/approve

Sorry for delay on approving this. The proposal is strictly specific to AWS and @gjtempleton is an owner of AWS cloudprovider, so I think his opinion (and not mine) is really the one that is relevant here. That being said this proposal looks very reasonable to me.

File: cluster-autoscaler/proposals/circumvent-tag-limit-aws.md (hunk 1 above)
Comment (MyannaHarris) [reply cluster-autoscaler/proposals/circumvent-tag-limit-aws.md:130]: Thank you for taking a look at my proposal.
There's no precedent for using parallel goroutines to do similar types of API calls in the AWS cloud-provider code. I'm less familiar with the rest of the code, but I don't think there's a precedent in it either.

Comment (MaciekPytel) [reply cluster-autoscaler/proposals/circumvent-tag-limit-aws.md:130]: There is a precedent in GCE provider: https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/cloudprovider/gce/cache.go#L277.

**Pull request merged**